test/functional/rpc_packages.py

```diff
@@ -306,5 +306,80 @@ def test_rbf(self):
+        results and getmempoolentry info. Note that the result structs are different and, due to
+        policy differences between testmempoolaccept and submitpackage (i.e. package feerate),
+        some information may be different.
+        """
+        for testres_tx in testres_package:
+            # Grab this result from the submitres
+            submitres_tx = submitres_package["tx-results"][testres_tx["wtxid"]]
+            assert_equal(submitres_tx["txid"], testres_tx["txid"])
+            if "allowed" in testres_tx and testres_tx["allowed"]:
+                assert_equal(submitres_tx["vsize"], testres_tx["vsize"])
+                assert_equal(submitres_tx["fees"]["base"], testres_tx["fees"]["base"])
+            entry_info = node.getmempoolentry(submitres_tx["txid"])
+            assert_equal(submitres_tx["vsize"], entry_info["vsize"])
+            assert_equal(submitres_tx["fees"]["base"], entry_info["fees"]["base"])
+
+    def test_submit_child_with_parents(self, num_parents):
+        node = self.nodes[0]
+        # Test a package with num_parents parents and 1 child transaction.
+        package_hex = []
+        package_txns = []
+        values = []
+        scripts = []
+        for _ in range(num_parents):
+            parent_coin = self.coins.pop()
+            value = parent_coin["amount"]
+            (tx, txhex, value, spk) = make_chain(node, self.address, self.privkeys, parent_coin["txid"], value)
+            package_hex.append(txhex)
+            package_txns.append(tx)
+            values.append(value)
+            scripts.append(spk)
+        child_hex = create_child_with_parents(node, self.address, self.privkeys, package_txns, values, scripts)
+        package_hex.append(child_hex)
+        package_txns.append(tx_from_hex(child_hex))
+
+        testres_package = node.testmempoolaccept(rawtxs=package_hex)
+        submitres_package = node.submitpackage(package=package_hex)
+
+        # Check that each result is present, with the correct size and fees
+        for i in range(num_parents + 1):
+            tx = package_txns[i]
+            wtxid = tx.getwtxid()
+            assert wtxid in submitres_package["tx-results"]
+            tx_result = submitres_package["tx-results"][wtxid]
+            expected_fee = Decimal("0.0001")
+            assert_equal(tx_result, {
+                "txid": tx.rehash(),
+                "vsize": tx.get_vsize(),
+                "fees": {
+                    "base": expected_fee,
+                }
+            })
+
+        # submitpackage result should be consistent with testmempoolaccept and getmempoolentry
+        self.assert_equal_package_results(node, testres_package, submitres_package)
+
+        # With no deduplications, the package feerate is just the ancestor feerate of the child.
+        child_txid = testres_package[-1]["txid"]
+        if "package-feerate" in submitres_package:
```

Comment:
Is this always true?

Reply:
Not if we have partial submissions - added that test case and an explanation comment